packages/block-library/src/image/image.js

```diff
@@ -172,6 +172,17 @@ export default function Image( {
 		} );
 	}
 
+	function uploadExternal() {
+		window
+			.fetch( url )
```

Comment:
Looks like this won't won't work because of CORS. It will have to happen through the server.